### PR TITLE
EDM-656: agent/device improve systemd status reporting

### PR DESCRIPTION
**Pull request by hexfusion**
This PR improves systemd status reporting. Although in the future it will have its own status this improves the interim reporting.

```
  status:
    applications:
    - name: oci-demo-app
      ready: 3/3
      restarts: 0
      status: Running
    - name: chronyd.service
      ready: 1/1
      restarts: 0
      status: Running
 ```

### CODE REVIEW

**Comment by hexfusion** [comment]
cc @celdrake

**Review by celdrake — COMMENTED** [review]
LGTM. 

Not a blocker for this PR, but I will discuss with Nir regarding the systemd units, as we never had a proper UX review for them. ATM the UI only supports operations for systemd units of fleetless devices.

File: internal/agent/device/device.go [hunk 1]
```diff
@@ -111,7 +118,11 @@ func (a *Agent) Run(ctx context.Context) error {
 
 // Stop ensures that the device agent stops reconciling during graceful shutdown.
 func (a *Agent) Stop(ctx context.Context) error {
-	a.cancelFn()
+	a.once.Do(func() {
```
Comment (hexfusion) [inline internal/agent/device/device.go:121]: note: although the agent is not expected to be thread-safe we need to ensure shutdown is.

**Comment by hexfusion** [comment]
> LGTM.
> 
> Not a blocker for this PR, but I will discuss with Nir regarding the systemd units, as we never had a proper UX review for them. ATM the UI only supports operations for systemd units of fleetless devices.

Yeah its an interim state but better than what it is. We will look at prioritizing the systemd status changes. But we should merge this for now.

File: internal/agent/device/device.go [hunk 2]
```diff
@@ -339,8 +350,22 @@ func (a *Agent) syncDevice(ctx context.Context, current, desired *v1alpha1.Rende
...
-	// set status collector properties based on new desired spec
-	a.statusManager.SetProperties(desired)
+	if err := a.systemdControllerSync(ctx, desired); err != nil {
+		return fmt.Errorf("systemd: %w", err)
+	}
+
+	return nil
+}
+
+func (a *Agent) systemdControllerSync(_ context.Context, desired *v1alpha1.RenderedDeviceSpec) error {
```
[the remark this answers is not in the document]
Comment (hexfusion) [reply internal/agent/device/device.go:360]: Right, its not used but intended to continue the pattern as this is more or less a transient state before we move to full systemd support which will use the ctx.

**Pull request merged**